cdk/src/open-data-platform/data-plane/data-import/data-import-stack.ts

```diff
@@ -67,6 +71,12 @@ export class DataImportStack extends Construct {
         let role = iam.Role.fromRoleArn(scope, id + '-' + f, f.role.roleArn);
         cluster.grantDataApiAccess(role);
       }
+
+      // Trigger all the data import lambdas when files change.
+      const destination = new notification.LambdaDestination(f);
```

Comment:
I wonder if we can also match on filename here to update on like an OBJECT_MODIFIED event - like if f.s3params.key == modified filename?

Not urgent just a thought, fine to leave a TODO

Reply:
It seems like they only have created and deleted events which is unfortunate. So we need to make sure we delete it temporarily and then re-upload https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-how-to-event-types-and-destinations.html

Reply:
I updated this to use a mapping between the ids for the lambda functions : the s3 files. There's some remaining TODOs but now the functions trigger only on relevant files.